lark/parsers/earley.py

```diff
@@ -37,18 +39,11 @@ def __init__(self, parser_conf, term_matcher, resolve_ambiguity=True, debug=Fals
         self.TERMINALS = { sym for r in parser_conf.rules for sym in r.expansion if sym.is_term }
         self.NON_TERMINALS = { sym for r in parser_conf.rules for sym in r.expansion if not sym.is_term }
 
-        self.forest_sum_visitor = None
+        self.forest_sum_visitor = ForestSumVisitor
         for rule in parser_conf.rules:
             if rule.origin not in self.predictions:
                 self.predictions[rule.origin] = [x.rule for x in analysis.expand_rule(rule.origin)]
 
-            ## Detect if any rules have priorities set. If the user specified priority = "none" then
-            #  the priorities will be stripped from all rules before they reach us, allowing us to
-            #  skip the extra tree walk. We'll also skip this if the user just didn't specify priorities
-            #  on any rules.
-            if self.forest_sum_visitor is None and rule.options.priority is not None:
-                self.forest_sum_visitor = ForestSumVisitor
```

Comment:
Wasn't this put here for optimization, for when there were on priorities? Is it not relevant anymore?

Reply:
Yes, it is an optimization. It can still be applied, but we also need to check that none of the terminals have priorities. However, the default terminal priority is 1 which is a meaningful value to the `ForestSumVisitor`. In other words, if a terminal has a priority of 1, we do not know whether or not it was given a priority in the grammar. If we want to implement this optimization, my suggestion would be change the default terminal priority to 0 (which would also be consistent with the default rule priority of 0).

Reply:
I understand. Then perhaps we should check if the priority != 1 for now?

I agree it would make sense to set both priorities to 0. Unless we actually want to choose derivations with more tokens than rules? I don't really know.

Anyway, we'll be able to change it to 0 in v1.0 which is coming up soon. So maybe we should wait until then?

Reply:
Checking priority != 1 is an option. Although, as you mentioned, the default of 1 causes the selection of derivations with more tokens. So either way we are breaking behavior.

Reply:
Is it breaking? Do we currently specify which one gets selected without any other specification?

Reply:
No, we don't specify which one is selected. So, in that sense it is not breaking, but perhaps an unexpected change in behavior.